# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/vm/Shape-inl.h
# Commit: 40d2e0691861
# Full Hash: 40d2e069186171b8b1389e3388fad5927accbd7e
# Author: Jan de Mooij <jdemooij@mozilla.com>
# Date: 2021-03-06 09:33:16
# Regressor Bug: 1689413
# File Overlap Count: 1
# Description:
#   Bug 1689413 part 10 - Stop passing proto in a few places now that we can get it from the shape. r=jonco
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D106980
# ==============================================================================

diff -r ec72f104edbf -r 40d2e0691861 js/src/vm/Shape-inl.h
--- a/js/src/vm/Shape-inl.h	Sat Mar 06 01:05:22 2021 +0000
+++ b/js/src/vm/Shape-inl.h	Sat Mar 06 01:05:23 2021 +0000
@@ -265,8 +265,7 @@
 
   // Cache the initial shape for non-prototype objects, however, so that
   // future instances will begin life with that shape.
-  RootedObject proto(cx, obj->staticPrototype());
-  EmptyShape::insertInitialShape(cx, shape, proto);
+  EmptyShape::insertInitialShape(cx, shape);
   return true;
 }
 